File: src/LCR018.XltzEq.1000254/solution.java
class Solution {
    public boolean isPalindrome(String s) {
        int left = 0;
        char[] chars = s.toLowerCase().toCharArray();
        int right = s.length()-1;
        while(left<right){
            while (left<right && noValid(chars[left])){
                left++;
            }
            while (left<right && noValid(chars[right])){
                right--;
            }
            if(left<right && chars[left] != chars[right]){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }

    private boolean noValid(char c){
        return !Character.isDigit(c) && !Character.isAlphabetic(c);
    }
}

File: src/LCR018.XltzEq.1000254/solution.ts
function isPalindrome (s: string): boolean {
    s = s.toLowerCase();
    let left = 0;
    let right = s.length - 1;
    while (left < right) {
        if (!isValidCode(s.charCodeAt(left))) {
            left++;
            continue;
        }
        if (!isValidCode(s.charCodeAt(right))) {
            right--;
            continue;
        }
        if (s[left] !== s[right]) {
            return false;
        }
        left++;
        right--;
    }
    return true;
}

function isValidCode (code:number) {
    return (code > 47 && code < 58) || (code > 96 && code < 123);
}
